docs/book/src/clusterctl/configuring-clusterctl.md

```diff
@@ -0,0 +1,46 @@
+# The clusterctl config file
+
+The `clusterctl` config file is should be located at `$HOME/.cluster-api/clusterctl.yaml` and it can be used for:
+
+- Customizing the list of providers and provider repositories that `clusterctl` can use
+- Providing values to be used for variable substitution when installing providers or creating clusters
+
+## Provider repositories
+
+The `clusterctl` command is designed to work with all the providers implementing
+the [clustetrctl contract](contract.md).
+
+Each provider is expected to define a provider repository, that is well-known place where the release assets for 
+a provider are published. 
+
+The `clusterctl` command ships with a pre-defined list of provider repositories that includes all the providers
+sponsored by SIG-cluster-lifecycle. 
+
+The user can change this list by adding new providers or changing the
+repository address for a pre-defined provider by editing the `clusterctl` config file as in the following example.
+
+```yaml
+providers:
+  # add a custom provider
+  - name: "my-infra-provider"
+    url: "https://github.com/myorg/myrepo/releases/latest/infrastructure_components.yaml"
+    type: "InfrastructureProvider"
+  # override a pre-defined provider
+  - name: "cluster-api"
+    url: "https://github.com/myorg/myforkofclusterapi/releases/latest/core_components.yaml"
+    type: "CoreProvider"
+```
+
+## Variables
+
+`clusterctl` supports variable substitution both in the components YAML and in the cluster template YAML.
```

Comment:
We should add a section above explaining in more details what we're trying to accomplish with variables. For new users "components yaml" or "cluster template yaml" are probably new concepts, and it might be a little early here to mention them